seed-job/template.yaml

```diff
@@ -158,8 +164,10 @@ objects:
                         // we need to make sure the image is removed in either case
                         stage('Remove the image') {
                           // don't fail the pipeline if image removal failed due to some error
-                          // the dangling images removal should be handled seprately by some cron etc
                           sh (script: "docker rmi ${image_name_with_registry}; docker rmi ${image_name}", returnStatus: true)
+                          def output = sh(returnStdout:true, script: "docker rmi ${base_image_name} > image-remove 2>&1", returnStatus: true)
+                          sh "cat image-remove"
+                          sh(script: "docker rmi `docker images -q -f dangling=true`", returnStatus: true)
```

Comment:
One more thing here is that,

these two steps will remove 
1. dangling image(s)
2. base image
in cases for
1. whether (new) base image is pulled
2. base image is not pulled

then, 
we don't need the earlier step for removal of base_image explicitly, as `--pull` option is anyway taking care of pulling the right one.

This is remove unnecessary step in the flow.

Reply:
@dharmit WDYT?

Reply:
there are two things.
1. For the first removal of image: 
     a. If the image is existing there, it will remove the image so that a new dangling image is not created.
     b. If the image is not existing or can not be removed because it is being used by some other image, command will simply return true, will not do anything.

2. For the second image removal:
    a. If the base image is pulled ( most of the times this will be the case, as we are removing the base image before pulling), we should remove it. This will make sure images are always built with base image pulled from registry.
   b. If the base image is not pulled, then also it should be removed to make sure the image is not left over after the build.

So even though docker `--pull` option always pulls the correct image. IMO, there is no reason to let docker create a dangling image.

Reply:
Ok so, there are other images like `ci-slave` getting  removed as dangling images. What I am not able to understand is, what harm it will cause us if we remove the base image before build to have a clean slate. So there will not be any chance of building on top of and old image.

Anyway, I am happy to remove that image deletion step to get this closed quickly(As we need this ASAP). we may have to add it back, if we move to buildah or something does not support force pull.

WDYT @navidshaikh @dharmit

Reply:
> there are other images like ci-slave getting removed as dangling images.

why so?

Reply:
> What I am not able to understand is, what harm it will cause us if we remove the base image before build to have a clean slate.

The point is, it's removal or existence is not affecting the current build. 

Case 1: Base image is updated at registry
  - the right base image is pulled and the resultant image is rebuilt on top of correct base image
  - a dangling base image is left behind (which will be cleared in later stage)

Case 2: Base image is not updated at registry
 - *the docker build step will not pull base image again (minimizing build time)*

Reply:
> > there are other images like ci-slave getting removed as dangling images.
> 
> why so?

Did not do a RCA, but may be because we have a force pull in seed-job and master-job for the `ci-slave` image

Reply:
removed the first image removal